ukis_pysat/data.py

```diff
@@ -43,16 +44,22 @@ class Source:
     def __init__(self, datahub, catalog=None):
         """
         :param datahub: Data source (<enum 'Datahub'>).
-        :param catalog: Path to catalog.json that holds the metadata if datahub is 'STAC' (String, Path).
+        :param catalog: Only applicable if datahub is 'STAC'. Can be one of the following types.
+                        Path to catalog.json that holds the metadata (String, Path).
+                        Pystac Catalog or Collection object (pystac.catalog.Catalog, pystac.collection.Collection).
+                        None initializes an empty catalog.
         """
         self.src = datahub
 
         if self.src == Datahub.STAC:
-            if not isinstance(catalog, (str, Path)):
-                raise AttributeError(f"'catalog' has to be set if datahub is 'STAC'.")
+            # connect to Spatio Temporal Asset Catalog
+            if isinstance(catalog, (pystac.catalog.Catalog, pystac.collection.Collection)):
+                self.api = catalog
+            elif isinstance(catalog, (str, Path)):
+                href = Path(catalog).resolve().as_uri()
+                self.api = pystac.catalog.Catalog.from_file(href)
             else:

```

Comment:
It's a bit confusing when catalog is not `None` and ends up here, I would be better to add another step here.